### DISCUSSION TITLE
Add Note Box for File Uploads

### DISCUSSION THREAD

**Discussion by NicoPowers** [cloudscape-design/components#2016] [Q&A] [answered]
Hi all,

I have been using the File upload component, and so far it has been great! All the UI components have been slick and love this design system!

One thing I have been wanting to do though, is to add a description box or note for each file upload, so a customer can quickly write a little note, if needed, for each file they uploaded.

Is there any way to achieve this? Would I have to make a custom wrapper component that uses the File component internally? I would like to use the `multiple` prop from the File upload, so doing it manually like this would seem to be an antipattern as I would be fighting the system.

Any guidance on this is greatly appreciated!

Thank you!

Nicolas

**Comment by jperals** [comment, marked as answer]
Hi Nicolas,

Thank you for your request, and glad that you are enjoying using Cloudscape! Currently there is no way to do exactly what you describe with the File Upload component, and there are no plans to add such a feature.

As you suggest, it would be possible to come up with a higher order component where multiple instances of the File Upload component could be revealed as the user is offered the option of uploading more files. Each of these instances could eventually use the `multiple` prop as well; however, it would not be possible for users to label each individual file when uploading multiple at once with a single instance of the component. Also, it could be confusing to have two different ways to upload multiple files. I think this could make sense if you want users to be able to group file uploads by some sort of conceptual categories that you can label, but I don't know if that is your case.

Hopefully this helped clarify, or at least confirm what you already thought.
